src/pnpq/devices/switch_thorlabs_osw1310e.py

```diff
@@ -1,48 +1,182 @@
-#
-# Thorlanbs Oprical Switch 1x2 and 2x2 1310E driver
-#       OSW12-1310E & OSW22-1310E
-#
+import time
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from threading import Lock
+from types import TracebackType
+
 import serial
 import serial.tools.list_ports
 from serial import Serial
 
+from .utils import timeout
+
+
+class State(Enum):
+    BAR = 1
+    CROSS = 2
+
+
+class AbstractOpticalSwitchThorlabs1310E(ABC):
+    @abstractmethod
+    def set_state(self, state: State) -> None:
+        """Set the switch to the specified state."""
+
+    @abstractmethod
+    def get_status(self) -> State:
+        """Get the current state of the switch."""
+
+    # Get system information
+    @abstractmethod
+    def get_query_type(self) -> str:
+        """Get the query type of the switch."""
+
+    @abstractmethod
+    def get_board_name(self) -> str:
+        """Get the board name of the switch."""
+
+    @abstractmethod
+    def open(self) -> None:
+        """Open the serial connection to the switch."""
+
+    @abstractmethod
+    def close(self) -> None:
+        """Close the serial connection to the switch."""
```

Comment:
I'm not sure if these methods should be part of the API contract or not. On the one hand, it's easily possible to implement an optical switch that doesn't use the context-manager paradigm for opening and closing. On the other hand, it would make things easier if we can assume that all stub implementations can be used as context managers.

So, that being said, if we're going to make `open` and `close` part of the API contract, we should make `__enter__` and `__exit__` part of it as well.